src/hooks/useDropzoneArea.js

```diff
@@ -0,0 +1,80 @@
+import {useState, useEffect, useCallback} from 'react';
+import {createFileFromUrl, readFile} from '../helpers';
+
+export const useDropzoneArea = ({
+    onChange,
+    clearOnUnmount,
+    initialFiles,
+    filesLimit,
+    onDelete}) => {
+    const [fileObjects, setFileObjects] = useState([]);
+
+    useEffect(() => {
+        if (onChange) {
+            onChange(fileObjects.map((fileObject) => fileObject.file));
+        }
+    }, [fileObjects, onChange]);
+
+    useEffect(() => {
+        loadInitialFiles();
+
+        return () => {
+            if (clearOnUnmount) {
+                setFileObjects([]);
+            }
+        };
+    }, [loadInitialFiles]);
+
+    const loadInitialFiles = useCallback(async() => {
+        try {
+            const fileObjs = await Promise.all(
+                initialFiles.map(async(url) => {
+                    const file = await createFileFromUrl(url);
+                    const data = await readFile(file);
+                    return {
+                        file,
+                        data,
+                    };
+                })
+            );
+
+            setFileObjects((prev) => ([...prev, ...fileObjs]));
+        } catch (err) {
+            console.log(err);
+        }
+    }, [initialFiles]);
+
+    const addFiles = useCallback(async(newFileObjects) => {
```

Comment:
Doesn't seem like this needs to be async

Reply:
I have been looking in the code history and this function has been async quite a while, I'm not sure I want to change this as part of a refactor task, perhaps we could investigate this separately whether this could be made synchronous and raise a separate issue for it. I think I saw that making one of the actions async fixed a bug somewhere, although I can't find it at the moment.

Reply:
I have created Issue #219  to look at this